internal/workload/podman/podman.go

```diff
@@ -316,3 +323,56 @@ func (p *podman) GenerateSystemdService(podId string, monitoringInterval uint) (
 
 	return svc, nil
 }
+
+// Retrieve all pods logs and send that to the given io.Writer
+func (p *podman) Logs(podID string, res io.Writer) (context.CancelFunc, error) {
+
+	podInfo, err := pods.Inspect(p.podmanConnection, podID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := containers.LogOptions{
+		Follow: &boolTrue,
+		Stderr: &boolTrue,
+		Stdout: &boolTrue,
+		Tail:   &logTailLines,
+	}
+
+	ctx, cancel := context.WithCancel(p.podmanConnection)
+	stdoutCh := make(chan string, 100)
+	stderrCh := make(chan string, 100)
+
+	for _, container := range podInfo.Containers {
```

Comment:
This loop starts logs streaming for all containers using one set of channels and as a result the log lines from multiple containers will be mixed in one writer (possibly even not monotonically). 

Do I understand the code correctly? If so, I don't think that is desired.

Reply:
yes, the logs are per pod, how many use cases we're going to have of multiple containers in each edge device? should we add more logic in there per container, etc..? I do not think at this stage. We can return an array of io.Reader, but I'm not sure if it's going to be helpful at all. 

Saying that, Podman added a pause container too, but as I said, multiple readers can be returned.

Reply:
I think we should handle multi-container pods correctly from the start. It could be returning multiple readers or annotating each log line with container ID. I would prefer feeding the logs into separate structures that could easily be transformed into files.

Reply:
Ok